torchmultimodal/models/clip/text_encoder.py

```diff
@@ -69,6 +69,11 @@ def __init__(
         self.ln_final = Fp32LayerNorm(width)
         self.projection = nn.Linear(width, embedding_dim, bias=False)
 
+        self.mask = torch.full(
```

Comment:
Once you use SDPA directly you should be able to avoid allocating this entirely. If you need to migrate checkpoints you could look into using a hook. For example the following code concatenates three separate weights into a single one for input projections of nn.MHA (courtesy of LLaMa).

```
    def __init__(
        self,
        dim: int,
        head_dim: int,
        n_heads: int,
        n_kv_heads: int,
    ):
        [...]
        self._register_load_state_dict_pre_hook(self.load_hook)

    def load_hook(
        self,
        state_dict,
        prefix,
        local_metadata,
        strict,
        missing_keys,
        unexpected_keys,
        error_msgs,
    ):
        if prefix + "wq.weight" in state_dict:
            wq = state_dict.pop(prefix + "wq.weight")
            wk = state_dict.pop(prefix + "wk.weight")
            wv = state_dict.pop(prefix + "wv.weight")
            state_dict[prefix + "wqkv.weight"] = torch.cat([wq, wk, wv])
```